modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/notifications/data/NotificationsPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,211 @@
+                            add(
+                                NotificationPreference.RadioGroupPreference(
+                                    title = context.getString(LR.string.settings_notification_vibrate),
+                                    value = settings.notificationVibrate.value,
+                                    options = listOf(
+                                        NotificationVibrateSetting.NewEpisodes,
+                                        NotificationVibrateSetting.OnlyWhenSilent,
+                                        NotificationVibrateSetting.Never,
+                                    ),
+                                    preference = NotificationPreferences.NEW_EPISODES_VIBRATION,
+                                    displayText = context.getString(settings.notificationVibrate.value.summary),
+                                ),
+                            )
+                        }
+                    }
+                },
+            ),
+            NotificationPreferenceCategory(
+                title = context.getString(LR.string.settings),
+                preferences = listOf(
+                    NotificationPreference.RadioGroupPreference(
+                        title = context.getString(LR.string.settings_notification_play_over),
+                        value = settings.playOverNotification.value,
+                        preference = NotificationPreferences.SETTINGS_PLAY_OVER,
+                        options = listOf(
+                            PlayOverNotificationSetting.NEVER,
+                            PlayOverNotificationSetting.DUCK,
+                            PlayOverNotificationSetting.ALWAYS,
+                        ),
+                        displayText = context.getString(settings.playOverNotification.value.titleRes),
+                    ),
+                    NotificationPreference.SwitchPreference(
+                        title = context.getString(LR.string.settings_notification_hide_on_pause),
+                        value = settings.hideNotificationOnPause.value,
+                        preference = NotificationPreferences.SETTINGS_HIDE_NOTIFICATION_ON_PAUSE,
+                    ),
+                ),
+            ),
+        )
+    }
+
+    private suspend fun getPodcastsSummary() = podcastManager.findSubscribedFlow().map { podcasts ->
+        val podcastCount = podcasts.size
+        val notificationCount = podcasts.count { it.isShowNotifications }
+
+        when {
+            notificationCount == 0 -> context.getString(LR.string.settings_podcasts_selected_zero)
+            notificationCount == 1 -> context.getString(LR.string.settings_podcasts_selected_one)
+            notificationCount >= podcastCount -> context.getString(LR.string.settings_podcasts_selected_all)
+            else -> context.getString(
+                LR.string.settings_podcasts_selected_x,
+                notificationCount,
+            )
+        }
+    }.take(1).firstOrNull()
```

Comment:
No need to call `take(1)` here. Also, it would be more appropriate to use `first()` instead of `firstOrNull()`, since we’re fetching a list and it’s never going to be `null`. If there are no subscribed podcasts, the list will just be empty.